docs/source/methodology.rst

```diff
@@ -1,6 +1,98 @@
+
+When starting an ``mcy`` project, it is recommended to start by using this "naive" logic directly, and keeping the number of mutations small. As you gain confidence that the setup is working, you can progressively add optimizations that let you run larger numbers of mutations.
+
+The ``bitcnt`` example shows an implementation of the naive logic:
+
+.. code-block:: text
+
+	sim_okay = result("test_sim") == "PASS"
+	eq_okay = result("test_eq") == "PASS"
+
+	if sim_okay and not eq_okay:
+	    tag("UNCOVERED")
+	elif not sim_okay and not eq_okay:
+	    tag("COVERED")
+	elif sim_okay and eq_okay:
+	    tag("NOCHANGE")
+	elif not tb_okay and eq_okay:
+	    tag("EQGAP")
+	else:
+	    assert 0
+
+This unconditionally runs both the test suite (which consists of a single testbench ``test_sim``), and the equivalence check ``test_eq``. The first three cases are the expected ones: if the equivalence check says there is a relevant difference in behavior between the original and mutated designs, then if the testbench catches it the mutation is covered, and if it fails to catch it it is not covered. If the equivalence check and the testbench both agree that the original and mutated designs behave the same way, the mutation is not useful for evaluating testbench quality and can be ignored.
+
+Keep an eye out for that fourth category! In a correctly set up project, this should never happen. If the equivalence check finds that the mutation does not affect the behavior, but the testbench finds a bug, one of them is wrong. Either the equivalence check is failing to find relevant behavioral changes, or the testbench is failing compliant designs. Investigate these cases thoroughly and fix the problem. Only proceed to optimize the logic once you are not observing any cases of the fourth type.
+
+For performance reasons, you will often want to end up at a logically equivalent but very different flow in the ``[logic]`` section, and this can be a source of confusion when looking at more complex, optimized examples. There are two assumptions that can be used to avoid running some of the tests, and still arrive at the same result:
+
+.. topic:: Assumption 1: The equivalence check is correct.
+
+	If the equivalence check fails, the mutation changes the functioning of the design, if it passes, there is no functional difference between the original and the mutated design.
+
+In particular, this means that if the equivalence check passes, there is no need to run any of the test suite on this mutation. Since the mutation does not have any relevant effect, it cannot tell us anything about the test suite's ability to detect bugs.
+
+.. topic:: Assumption 2: The test suite will not return false failures for a working design.
+
+	It will not fail randomly, and it will not flag benign changes in behavior that conform to the specification.
```

Comment:
The title "The test suite will not return false failures for a working design" is easier to read and understand than the explanation below it.

Reply:
I think it's this part: it will not flag benign changes in behaviour that conform to the specification.
What is the specification?

Reply:
Yeah, I guess I shouldn't get philosophical about what it means for a design to be "correct" here! For all practical purposes, "if you find a gap, figure out if the behavior of the mutated design is ok or not, and adjust either the testbench or the equivalence check accordingly" will serve as a rule. How to know if it's ok can remain a "you know it when you see it" thing...

Reply:
Is this clearer?

>Assumption 2: The test suite will not return false failures for a working design.
>	Ensure that your testbenches are tolerant of small changes in behavior that are not relevant to correctness, such as e.g. the values of output bits that are unused in certain modes.